### PR TITLE
fix: remove bignumber from the redux store

### PR DESCRIPTION
**Pull request by victorkirov**
# 🔘 PR Type
What kind of change does this PR introduce?

- [x] Bugfix

# 📜 Background
When changing settings, if you have multiple tabs open, the saved settings are overwritten by the last tab that is closed, and settings are not propagated across tabs. This is due to the store sync not syncing everything, which was disabled because we had class instances in the store.

# 🔄 Changes
All BigNumber instances were removed from the store and the values were cast to BigNumber where needed throughout the code. A number of other components were also changed to fix any linting and typing issues that were blocking commits.

Impact:
- This changes stx and btc balances and fiat rates in the store, so it would affect any component which uses those 
- Since we updated the store sync, this will also affect onboarding and the data that was synced between tabs during onboarding

# 🖼 Screenshot / 📹 Video
Before:

https://github.com/secretkeylabs/xverse-web-extension/assets/3322667/772f4441-a74c-4000-b9e3-d6a83d222992

After:

https://github.com/secretkeylabs/xverse-web-extension/assets/3322667/cd594851-0ba4-4346-a0fd-fcd659c06e2f

# ✅ Review checklist
Please ensure the following are true before merging:

- [ ] Code Style is consistent with the project guidelines.
- [ ] Code is readable and well-commented.
- [ ] No unnecessary or debugging code has been added.
- [ ] Security considerations have been taken into account.
- [ ] The change has been manually tested and works as expected.
- [ ] Breaking changes and their impacts have been considered and documented.
- [ ] Code does not introduce new technical debt or issues.

### CODE REVIEW

File: src/app/screens/restoreFunds/restoreBtc/index.tsx [hunk 1]
```diff
@@ -1,28 +1,22 @@
+import IconBitcoin from '@assets/img/dashboard/bitcoin_icon.svg';
```
Comment (yknl) [inline src/app/screens/restoreFunds/restoreBtc/index.tsx:1]: I believe we don't need this screen anymore after the rare sats update.

File: src/app/screens/sendBrc20/index.tsx [hunk 2]
```diff
@@ -152,29 +152,30 @@ function SendBrc20Screen() {
...
-          setAmountError(t('SEND.ERRORS.INSUFFICIENT_BALANCE'));
-        } else if (Number(err) === ErrorCodes.InSufficientBalanceWithTxFee) {
-          setAmountError(t('SEND.ERRORS.INSUFFICIENT_BALANCE_FEES'));
-        } else setAmountError(err.toString());
-      });
+      );
       navigate('/confirm-inscription-request', {
         state: {
-          brcContent: order.inscriptionRequest.files[0].dataURL,
+          // !NOTE: Typing below is broken. Not going to fix as this page is deprecated.
```
Comment (yknl) [inline src/app/screens/sendBrc20/index.tsx:158]: Let's take this opportunity to remove the unused code.

File: src/app/stores/wallet/reducer.ts [hunk 3]
```diff
@@ -46,13 +45,13 @@ const initialWalletState: WalletState = {
   seedPhrase: '',
   encryptedSeed: '',
   fiatCurrency: 'USD',
-  btcFiatRate: new BigNumber(0),
-  stxBtcRate: new BigNumber(0),
-  stxBalance: new BigNumber(0),
-  stxAvailableBalance: new BigNumber(0),
-  stxLockedBalance: new BigNumber(0),
+  btcFiatRate: '0',
```
Comment (yknl) [inline src/app/stores/wallet/reducer.ts:48]: If we some how end up with an invalid number, does the app crash while trying to parse to a BigNumber?

Comment (victorkirov) [reply src/app/stores/wallet/reducer.ts:50]: Nope, we check the btcFiatRate value to ensure it's a valid number before setting it. See the `numberLikeToStringOrThrow` function in the actionCreator.ts file

File: src/app/screens/sendBrc20/index.tsx (hunk 2 above)
Comment (victorkirov) [reply src/app/screens/sendBrc20/index.tsx:158]: Will do 😄

**Review by yknl — DISMISSED** [review]
Let's make sure to test the Ledger send BRC-20 flow given the typing issue.

**Comment by DuskaT021** [comment]
@victorkirov can we resolve this conflict, tnx

**Comment by blehitsahmad** [comment]
@victorkirov kindly resolve the conflicts so this PR can be picked up for testing too

**Comment by blehitsahmad** [comment]
This PR will be picked up after the [seed vault PR](https://github.com/secretkeylabs/xverse-web-extension/pull/510) is  merged -- as per the discussion with @victorkirov

**Comment by blehitsahmad** [comment]
Hey @victorkirov could you resolve this conflicts so the testing for this PR can be started?

**Comment by teebszet** [comment]
@blehitsahmad this is ready for retest

**Comment by blehitsahmad** [comment]
@DuskaT021 could you kindly take a look at the ledger BRC20 send flow for this PR -- I have tested the rest. The test cases can be found [here](https://www.notion.so/xverseapp/EXT-PR-610-a52a82940532409a93115d5f42e207b0?pvs=4)

**Comment by DuskaT021** [comment]
@victorkirov this is tested and can be merged

**Review by teebszet — APPROVED** [review]

**Pull request merged**